wordpress-mobile/WordPress-Android
# Site Creation: Injected UI Helper

Fixes #8865. This PR adds an injected UI helper that's used throughout the site creation flow. It currently only has 2 helpers, one for extracting the text from `UIString` and the other for updating visibility both of which is used fairly often in the site creation flow. A couple notes about this PR:

* Although this is currently being added for site creation flow, it's just a generic helper that can be used anywhere in the project.
* Strictly speaking, injecting this helper is not really a must. The main concern about generic functions in Kotlin is that they are `static` methods in JVM which means we can't easily mock them. Since these `UI` helpers are used in the view layer, we probably won't need to mock them anyway. Having said that it's a good rule of thumb to put these functions in an injected class because then we can mock them easily if needed and it doesn't really have a bigger footprint than a static method. It's slightly "harder" to use them in custom views (as can be seen in `SearchInputWithHeader`) since we'll pass them in a constructor, but that's not a big of a concern.
* The name `UiHelpers` might not be the best and I am happy to change it if there are any suggestions.

**To test**

I don't think there is anything specific to test in this PR. One quick check of the new site creation flow should be enough. One can access this flow by going to the site picker page, tapping on the `+` menu button and selecting `Create WordPress.com site` and following the steps until the domains page. (After that the site will be created which is not necessary)
